### PR TITLE
De-complexify code and use pipes more places

### PR DESCRIPTION
**Pull request by greyblue9**

### CODE REVIEW

File: src/use/use.py [hunk 1]
```diff
@@ -526,9 +496,7 @@ def _venv_is_win() -> bool:
 
 def _pebkac_no_version_hash(func=None, *, name: str, **kwargs) -> Union[ModuleType,Exception]:
     if func:
-        result = func(name=name, **kwargs)
-        if isinstance(result, ModuleType):
-            return result
+        return func(name=name, **kwargs)
```
Comment (amogorkon) [inline src/use/use.py:499]: This doesn't take into account that func could return an Exception, which should NOT be returned in this case.

File: tests/coverage_combine.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
... trim | add_suffix ';' | tee ./0_0__schema.sql 1>&2; tblidx=-1; ntbls=${#tables[@]}; while (( ++tblidx < ntbls )); do tbl="${tables[tblidx]}"; echo -E "  - Load data from table $tbl" 1>&2;  { echo -nE "INSERT OR IGNORE INTO $tbl "; sqlite3 "${db}" -tabs "select * from $tbl" -separator $'\xf8' -newline $'\f' | sed -r -e 's~'"'"'~\xf9~g;   :a s~(\xf8|^)([^\n\xf8]*)($|\xf8)~'"'"'\n    '"'"'\2'"'"',\n    \3~g; ta; s~\f~'"'"',\n), (\n    '"'"'~g; ta; s~(^|\n +\xf8)([^\n\xf8]+[^'"'"'\\\n\xf8])\n +'"'"'~\n    '"'"'\2'"'"',\n    ~g; ta; s~\xf8~'"'"'~g; s~, *\([^()a-zA-Z0-9_]*$~;\n~; s~^[^\n]*\n~VALUES (\n~; s~\xf9~'"'"' + "'"'"'" + '"'"'~g; ' | sed -r -e '\~^ +'"'"'(.*)'"'"' *$~{ N; \~\n +'"'"'~{ s~~,&~; }; }; \~^    +'"'"'[\t\r ]*$~{ d; ' -e '    ; };  ' | tr -s '\n' '\v' | sed -r -e ':a s~,([\t\n\v\r ]*)(;|$|\))~\1\2~g; ta;' | tr -s '\v' '\n'; } > "${dbidx}_${tblidx}_${tbl}.sql";  grep -qFe "VALUES" --text -hs -- "${dbidx}_${tblidx}_${tbl}.sql" || rm -vf -- "${dbidx}_${tblidx}_${tbl}.sql" 1>&2; done; done;  rm coverage2.db; for f in ./0_*.sql ./[1-9]*.sql; do fn="${f##*/}"; nme="${fn%.*}"; tbl="${nme##*[0-9]_}"; sqlite3 coverage2.db -init "$f" "; VACUUM; REINDEX;"; echo "[ $? ] $f"; done
```
Comment (amogorkon) [inline tests/coverage_combine.sh:3]: wtf :p

File: src/use/use.py [hunk 3]
```diff
@@ -957,62 +920,56 @@ def _load_venv_entry(name, module_path) -> ModuleType:
 
 @cache(maxsize=512, typed=True)
```
Comment (amogorkon) [inline src/use/use.py:921]: why maxsize? i'm betting a large system can easily hit that number..

File: src/use/use.py [hunk 4]
```diff
@@ -682,7 +648,7 @@ def isfunction(x: Any) -> bool:
 
 
 def ismethod(x: Any) -> bool:
-    return inspect.isfunction(x)
+    return inspect.ismethod(x)
```
Comment (amogorkon) [inline src/use/use.py:651]: let's remove ismethod, isfunction. if we import them directly, they end up as globals in use anyway..

**Review by amogorkon — DISMISSED** [review]

File: src/use/use.py (hunk 1 above)
Comment (greyblue9) [reply src/use/use.py:499]: It doesn't happen, though

File: tests/coverage_combine.sh (hunk 2 above)
Comment (greyblue9) [reply tests/coverage_combine.sh:3]: It's finding all python files and making fully qualified names for each, so all files can be included

File: src/use/use.py (hunk 3 above)
Comment (greyblue9) [reply src/use/use.py:921]: default wasn't big enough

File: src/use/use.py (hunk 4 above)
Comment (greyblue9) [reply src/use/use.py:651]: ok, I agree

File: src/use/use.py (hunk 1 above)
Comment (greyblue9) [reply src/use/use.py]: Addressed

**Pull request merged**